### ISSUE TITLE
Regression with wmctrl

### ISSUE THREAD

**Issue by saygo-png** [awesomeWM/awesome#3973]
Output of `awesome --version`:

```
awesome 4.3 (Too long)
 • Compiled against Lua 5.2.4 (running with Lua 5.2)
 • D-Bus support: ✔
 • execinfo support: ✔
 • xcb-randr version: 1.6
 • LGI version: [string "return require('lgi.version')"]:1: module 'lgi.version' not found:
	no field package.preload['lgi.version']
	no file '/nix/store/jcqim0zqs9rna01kl0vjr7iv6rp0x366-lua-5.2.4/share/lua/5.2/lgi/version.lua'
	no file '/nix/store/jcqim0zqs9rna01kl0vjr7iv6rp0x366-lua-5.2.4/share/lua/5.2/lgi/version/init.lua'
	no file '/nix/store/jcqim0zqs9rna01kl0vjr7iv6rp0x366-lua-5.2.4/lib/lua/5.2/lgi/version.lua'
	no file '/nix/store/jcqim0zqs9rna01kl0vjr7iv6rp0x366-lua-5.2.4/lib/lua/5.2/lgi/version/init.lua'
	no file './lgi/version.lua'
	no file '/nix/store/jcqim0zqs9rna01kl0vjr7iv6rp0x366-lua-5.2.4/lib/lua/5.2/lgi/version.so'
	no file '/nix/store/jcqim0zqs9rna01kl0vjr7iv6rp0x366-lua-5.2.4/lib/lua/5.2/loadall.so'
	no file './lgi/version.so'
	no file '/nix/store/jcqim0zqs9rna01kl0vjr7iv6rp0x366-lua-5.2.4/lib/lua/5.2/lgi.so'
	no file '/nix/store/jcqim0zqs9rna01kl0vjr7iv6rp0x366-lua-5.2.4/lib/lua/5.2/loadall.so'
	no file './lgi.so'
```

**How to reproduce the issue:**
Exact copy of the issue below with the same reproduction steps.
https://github.com/awesomeWM/awesome/issues/3047

I've downgraded my awesome to a version from 2021 and it works again
Don't mind the "lgi not found" error, it doesnt matter and is not the issue here.

**Comment by saygo-png** [comment]
Closing, because turns out this is not an awesome issue, but just nixos using the release version of awesome and not git. This is the fix for nixos 24.05

```nix
  services.xserver.windowManager.awesome = {
    enable = true;
    package = pkgs.awesome.overrideAttrs (previousAttrs: {
      patches =
        previousAttrs.patches
        ++ [
          (builtins.toFile "fixwmctrl.patch" ''
            diff --git i/ewmh.c w/ewmh.c
            index eea5a95..7e1354d 100644
            --- i/ewmh.c
            +++ w/ewmh.c
            @@ -461,7 +461,7 @@ ewmh_process_client_message(xcb_client_message_event_t *ev)
                         lua_State *L = globalconf_get_lua_State();
                         luaA_object_push(L, globalconf.tags.tab[idx]);
                         lua_pushstring(L, "ewmh");
            -            luaA_object_emit_signal(L, -1, "request::select", 1);
            +            luaA_object_emit_signal(L, -2, "request::select", 1);
                         lua_pop(L, 1);
                     }
                 }
          '')
        ];
    });
  };
```

**Comment by actionless** [comment]
yeah, "stable" version of awesomewm is broken by design, unfortunately: https://github.com/awesomeWM/awesome/issues/3937
